webui/react/src/omnibar/Omnibar.tsx

```diff
@@ -0,0 +1,45 @@
+import OmnibarNpm from 'omnibar';
+import React, { useCallback } from 'react';
+
+import { StoreAction, useStoreDispatch } from 'contexts/Store';
+import * as Tree from 'omnibar/tree-extension/index';
+import TreeNode from 'omnibar/tree-extension/TreeNode';
+import { BaseNode } from 'omnibar/tree-extension/types';
+import { isTreeNode } from 'omnibar/tree-extension/utils';
+
+import css from './Omnibar.module.scss';
+
+const Omnibar: React.FC = () => {
+  const storeDispatch = useStoreDispatch();
+
+  const hideBar = useCallback(
+    () => storeDispatch({ type: StoreAction.HideOmnibar }),
+    [ storeDispatch ],
+  );
+
+  const onAction = useCallback((item, query) => {
+    // ideally we could useRef here but can we w/o modifying the omnibar package?
+    const input: HTMLInputElement|null = document.querySelector('#omnibar input[type="text"]');
```

Comment:
any ideas on how we could avoid this? an option is to push a ref down to the omnibar package but that'd involve changes to omnibar package

Reply:
just noting what we talked:
- potentially building the string from tree node paths
- using the `value` prop for the omnibar

Reply:
> just noting what we talked:
> 
> potentially building the string from tree node paths

We can do this part. If we don't immediately need this part I think it'll be better if we do this
together with multi level query matching as I suspect these two would affect eachother.

> using the value prop for the omnibar

since we don't have access to let the omnibar's input update this value as the user is typing I couldn't find a good way to use this.

Reply:
I saved this idea in a code comment